src/librustc_error_codes/error_codes/E0749.md

```diff
@@ -11,6 +11,7 @@ trait MyTrait {
 impl !MyTrait for u32 {
     type Foo = i32; // error!
 }
+// impl !MyTrait for u32 {} // fix
```

Comment:
Instead of putting it inside the erroneous code example, it would be better to put it into another code block at the end like the other explanations do.

Reply:
Good idea, I applied the suggestion. Do we need `fn main() {}`?

Reply:
I never remember how rustdoc handles global attributes `#![...]`, so let's wait to see if the CI is happy and we'll know for sure. :)

Reply:
@GuillaumeGomez Should we remove it? `fn main`

Reply:
I guess we should indeed. Please also [squash](https://blog.guillaume-gomez.fr/Git-tips/1/2) your commits.

Reply:
I am doing most documentation stuff from github directly, I don't want to interact with many branches (~5) I have on rust.

Reply:
4 commits for 11 lines of uncontroversial documentation seems like super huge to me.

Reply:
Well, rust already have > 120,000 commits, adding 4 won't make a difference, I do think it is a lot. I will squash if it is required, otherwise I rather be lazy and not squash. :P